src/crud/useWavesurferRegions.tsx

```diff
@@ -41,8 +41,10 @@ export const parseRegionParams = (
 export const parseRegions = (regionstr: string) => {
   if (!regionstr) return { params: {}, regions: [] as IRegion[] } as IRegions;
   var segs = JSON.parse(regionstr);
-  if (segs.regions) segs.regions = JSON.parse(segs.regions);
-  else segs.regions = [];
+  if (segs.regions) {
+    if (typeof segs.regions == 'string' || segs.regions instanceof String)
+      segs.regions = JSON.parse(segs.regions);
+  } else segs.regions = [];
```

Comment:
This logic can fail. Frequently regionstr will be initialized to '{}' and not to an empty string. That will lead to line 44 throwing an error because segs has no region parameter.

Reply:
It doesn't throw.  It returns false.